apps/projects/routes.py

```diff
@@ -176,100 +377,814 @@ def get(self, project_id):
                 metadata:
                   type: object
-                  example: {...}
+                  properties:
+                    codec_name:
+                      type: string
+                      example: h264
+                    codec_long_name:
+                      type: string
+                      example: H.264 / AVC / MPEG-4 AVC / MPEG-4 part 10
+                    width:
+                      type: integer
+                      example: 640
+                    height:
+                      type: integer
+                      example: 360
+                    duration:
+                      type: string
+                      example: 300.014000
+                    bit_rate:
+                      type: string
+                      example: 287654
+                    nb_frames:
+                      type: string
+                      example: 7654
+                    r_frame_rate:
+                      type: string
+                      example: 24/1
+                    format_name:
+                      type: string
+                      example: mov,mp4,m4a,3gp,3g2,mj2
+                    size:
+                      type: string
+                      example: 14567890
+                mime_type:
+                  type: string
+                  example: video/mp4
+                create_time:
+                  type: object
+                  properties:
+                    $date:
+                      type: integer
+                      example: 1556853105063
+                original_filename:
+                  type: string
+                  example: video.mp4
                 client_info:
                   type: string
-                  example: PostmanRuntime/7.6.0
+                  example: Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:66.0) Gecko/20100101 Firefox/66.0
                 version:
                   type: integer
                   example: 1
                 parent:
                   type: object
                   example: {}
+                processing:
+                  type: boolean
+                  example: False
                 thumbnails:
                   type: object
                   example: {}
+                _id:
+                  type: object
+                  properties:
+                    $oid:
+                      type: string
+                      example: 5cbd5acfe24f6045607e51aa
         """
-        item = app.mongo.db.projects.find_one({'_id': format_id(project_id)})
-        if not item:
-            return not_found("Project with id: {} was not found.".format(project_id))
 
-        return Response(json_util.dumps(item), status=200, mimetype='application/json')
+        doc = app.mongo.db.projects.find_one({'_id': format_id(project_id)})
+        if not doc:
+            return not_found("Project with id: {} was not found.".format(project_id))
+        return json_response(doc)
 
     def put(self, project_id):
```

Comment:
I'm not sure this should be PUT, it doesn't replace the resource with a version you post, and also repeating same request isn't safe, POST would be better.
and what about concurrency control?

Reply:
@petrjasek the idea is that with PUT you edit a video without creating a new project and with POST you create a new one. 

@idsvn there are a lot of similar code in PUT and POST, reuse it.